src/main/java/com/google/sps/servlets/GetGameDialogue.java

```diff
@@ -26,19 +31,26 @@
 public final class GetGameDialogue extends HttpServlet {
   private static final String TEXT_TO_HTML = "text/html;";
   private static final String JSON_CONTENT_TYPE = "application/json";
-  private static final String SOFTWAREID = "software-engineering-0";
-  private static final String TESTCONTENT = "hello this is a test";
-
-  // TODO: GameStage will change and be query from a database that will be implemented
-  private GameStage currentGameStage = new GameStage(SOFTWAREID, TESTCONTENT);
+  private static final String LOGGED_OUT_EXEPTION = "It appears the user you have is not logged in";
+  private static final Gson gson = new Gson();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final PlayerDatabase playerDatabase = new PlayerDatabase(datastore);
+  private final GameStageDatabase gameStageDatabase = new GameStageDatabase(datastore);
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // you can only use the PlayerDatabase sucessfully if you are running this on a live server
+      // because it requires you to log in
+      GameStage currentGameStage =
+          gameStageDatabase.getGameStage(playerDatabase.getEntityCurrentPageID());
+      String dialogue = gson.toJson(currentGameStage.getContent());
 
-    Gson gson = new Gson();
-    String dialogue = gson.toJson(currentGameStage.getContent());
-
-    response.setContentType(JSON_CONTENT_TYPE);
-    response.getWriter().println(dialogue);
+      response.setContentType(JSON_CONTENT_TYPE);
+      response.getWriter().println(dialogue);
+    } catch (Exception LoggedOutException) {
```

Comment:
Now that there is a logged out exception class, we can use that instead of the more general Exception

Reply:
+1